### DISCUSSION TITLE
Generate_coarse AssertionError

### DISCUSSION THREAD

**Discussion by cloedev** [suno-ai/bark#217] [General]
Morning everyone.
I'm trying to generate a wav file using a custom voice. I think I've managed to create my custom npz correctly so I'm using the following code to create my custom speech:

```
from bark.api import generate_audio
from transformers import BertTokenizer
from bark.generation import SAMPLE_RATE, preload_models, codec_decode, generate_coarse, generate_fine, generate_text_semantic

# Enter your prompt and speaker here
text_prompt = "Hola, me llamo Jose"
voice_name = "C:\\Development\\AI\\bark\\bark\\assets\\prompts\\custom\\jose.npz" # use your custom voice name here if you have one
#voice_name = "es_speaker_8" # use your custom voice name here if you have one

# load the tokenizer
tokenizer = BertTokenizer.from_pretrained("bert-base-multilingual-cased")

# download and load all models
preload_models(
    text_use_gpu=True,
    text_use_small=False,
    coarse_use_gpu=True,
    coarse_use_small=False,
    fine_use_gpu=True,
    fine_use_small=False,
    codec_use_gpu=True,
    force_reload=False#,
    #path="models"
)

# simple generation
audio_array = generate_audio(text_prompt, history_prompt=voice_name, text_temp=0.7, waveform_temp=0.7)
```
I'm getting this error:

File "c:\Development\ai\bark\bark\create.py", line 37, in <module>
    x_coarse_gen = generate_coarse(
                   ^^^^^^^^^^^^^^^^
  File "C:\Users\JossSrf\AppData\Local\Programs\Python\Python311\Lib\site-packages\bark\generation.py", line 587, in generate_coarse
    round(x_coarse_history.shape[-1] / len(x_semantic_history), 1)
AssertionError

Any ideas?
Thank you!

**Comment by gkucsko** [comment]
what does you npz file look like? looks like it fails during validation for vocabulary checks etc

**Reply by cloedev** [reply]
![image](https://user-images.githubusercontent.com/54842982/235356106-71953b79-479c-48aa-a6eb-4d7c7c8b672e.png)
I don't know if this is meant to be like. What do you suggest? Creating a new npz?

**Reply by gkucsko** [reply]
hm yeah that seems weird. weird a dictionary of the entries and then save it with `bark.api.save_as_prompt`
